### ISSUE TITLE
Steam OTP codes are decoded wrong in this script

### ISSUE THREAD

**Issue by sigboe** [emlun/yubikey-oath-dmenu#24]
Yubikey supports Steam's non-standard otp codes, htey are displayed correctly when using Yubico Desktop Authenticator or `ykman oath accounts code steam`, even though the correct settings to add steam's OTP format is not shown in Yubico Desktop Authenticator, just adding it with the Issuer as steam, Desktop Authenticator will detect it is the steams format and it will be saved properly. Yubico added support for this, its just not advertised so much, because of the difficulty of getting the shared key.

Problem is that using this python script they are displayed as a 6 digit numerical code, instead of the 5 digit alphanumerical code

(on a rooted phone you can just find the secret code in clear text on your phones storage)
As a Linux user with an Android phone that is **NOT** rooted, it was cumbersome to fetch the secret code. I had to preform these steps:

* Install Steam from playstore
* Disable steamguard on my phone if its already active (don't need to disable it completely on the account, just the phone so it is not registered as being your steamguard authenticator, if your get codes on email you are good)
* Install JDK on the computer
* Install APKtool on the computer (not the tool that provides the command apk with a similar name)
* install adb on the computer (already had this)
* install jq on the computer (I already had this)
* download abe from this repo https://github.com/nelenkov/android-backup-extractor/releases/tag/master-20220409062712-07bb660
* enable adb on phone (I already had this)
* connect phone to computer and run `adb devices` and make sure it is authenticated
* download get_code.sh from this repo https://github.com/steamguard-totp/steamguard-shared-secret
* above script is broken, need to at least fix line 12 and 14, remove the surrounding spaces around the `=` sign to get the syntax correct.
* also comment out lines 3,4,5,6,7 if you don't have python2, the things done in python2 here are broken anyway
* the above script is still broken, but it will complete except the last steps which id just did manually, I think it was the two last lines, lines 83 and 85 which I did manually, you may want to just comment them out, spececially since they require python2 which we removed the check for
* this will backup the steam apk, uninstall it, patch the apk file to allow for backups to be made of the data, backup the data to your computer
* The script will pause and ask you to log into the Steam app and enable Steam Guard Mobile Authenticator, do so. I saved the recovery code in a safe encrypted vault. Maybe a safe place that doesn't require your yubikey to open up.
* last part of the prior step will ask you on your phone to enter a password you want to set, because if your phone is encrypted backups will be required to also have a password. Just pick any decently secure password you can remember, you can delete the file later anyway, maybe don't do this on a computer you don't trust
* lastly I used these commands: 
 ```
java -jar abe.jar unpack backup.ab backup.tar
#this command asks you for your password you set recently
tar -xf backup.tar
cat apps/com.valvesoftware.android.steam.community/f/Steamguard-* | jq
```
* this will show you your `shared_secret` code in the terminal in plain text (also the revocation code)
* In Yubico Desktop Authenticator( should work in the phone authenticator or ykman as well) I added a new account, Issuer has to be Steam, I suggest the Account name is your account name, Shared key is the shared_secret displayed in the terminal in the prior step. Require touch is optional, the other settings should be default, it should pick the correct settings automatically, which are not available for you to pick. 
* On your phone, in google play uninstall and reinstall Steam, you will be logged out of Steam Guard on your phone, and now the only location to get steam guard otp codes is your yubikey. This step is needed, because unless we do this, a person with physical access to your phone, and can guess or brute force your pin code can get the steam guard shared secret. Also if we don't do this it maybe will just happen next time the Steam android app is updated anyway

**Comment by sigboe** [comment]
@dainnilsson wants to confirm that the issue is not in this project before investigating upstream :+1:

**Comment by sigboe** [comment]
After bothering yubico again, they have confirmed, that the python libraries are working as intended, and this project needs to check if the code is a steam code and convert it. This is how they do it in the cli tool ykman

https://github.com/Yubico/yubikey-manager/blob/main/ykman/oath.py

**Comment by emlun** [comment]
Hi! Sorry for the delay. Yes, [upstream works as intended](https://github.com/Yubico/yubikey-manager/issues/502#issuecomment-1122566120), this is a missing feature in `yubikey-oath-dmenu`. I'll see about implementing this sometime soon-ish.

**Comment by emlun** [comment]
This should be fixed now in [release 0.14.0](https://github.com/emlun/yubikey-oath-dmenu/releases/tag/v0.14.0). Please try it out!

**Comment by sigboe** [comment]
@emlun Thank you, this works for me!
